cypress/integration/admin_portal/pages/enterprise_code_management.js

```diff
@@ -4,30 +4,85 @@ class CodeManagementPage {
   }
 
   getLabels(labelName) {
-    const labels = {Email: 1, Company: 2, Codes:3}
+    const labels = { Email: 1, Company: 2, Codes: 3 }
     return cy.get(`.form-group:nth-of-type(${labels[labelName]}) label`)
   }
 
-  getRequestCodesButton(){
+  getRequestCodesButton() {
     return cy.get('.btn.btn-primary')
   }
 
-  getCancelButton(){
+  getCancelButton() {
     return cy.get('.btn.btn-link.ml-3.form-cancel-btn')
   }
 
-  getInvalidFeedback(){
+  getInvalidFeedback() {
     return cy.get('[name="emailAddress"]~.invalid-feedback')
   }
 
-  getSuccessMessage(){
-      return cy.get('.alert-dialog .message')
+  getSuccessMessage() {
+    return cy.get('.alert-dialog .message')
   }
 
-  requestMoreCodes(){
+  requestMoreCodes() {
     cy.get('.request-codes-btn.btn.btn-primary').click()
   }
 
+  getCoupon() {
+    return cy.get('.coupon')
+  }
+
+  getCouponDetailsRow() {
+    return cy.get('.expanded .metadata.row')
+  }
+
+  getCouponDetailsRowText() {
+    return cy.get('small.text-light')
+  }
+
+  getFilterOptions() {
+    return cy.get('select[name="table-view"] option')
+  }
+
+  getCodeStatusFilter() {
+    return cy.get('select[name="table-view"]')
+  }
+
+  getBulkActionFilter() {
+    return cy.get('select[name="bulk-action"] option')
+  }
+
+  getAssignActionButton() {
+    return cy.get('.table tbody tr:nth-of-type(1) .btn')
+  }
+
+  getRemainingAssignments() {
+    return cy.get('.table tbody tr:nth-of-type(1) td:nth-of-type(4)')
+  }
+
+  getModalWindow() {
+    return cy.get('.modal-content')
+  }
+
+  getCodeAssignmentSuccessMessage() {
+    return cy.get('.alert-success .message')
```

Comment:
you can return the message instead of element

Reply:
I need the element as well to get the child. Doing this was is allowing me to get the text and the child in the same chainer.